From 645581a8b093e9d6aa039382c5a6434c65b06c78 Mon Sep 17 00:00:00 2001
From: jim <jim.yuan@supermicro.com>
Date: Thu, 7 Jul 2016 15:53:31 -0700
Subject: [PATCH] Hardcode DIMM pairing for RAS. Fix the issue that MBA0 Memory
 UE on Cenatur Chip4 brought all dimms under Proc2 offline.

Signed-off-by: Jim Yuan <jim.yuan@supermicro.com>
---
 src/usr/hwas/common/deconfigGard.C | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/src/usr/hwas/common/deconfigGard.C b/src/usr/hwas/common/deconfigGard.C
index fa266c8..bd4e4fe 100644
--- a/src/usr/hwas/common/deconfigGard.C
+++ b/src/usr/hwas/common/deconfigGard.C
@@ -980,8 +980,25 @@ Target * findPartnerForMcs(const Target *i_startMcs)
     // Declare partner MCS CHIP_UNIT
     ATTR_CHIP_UNIT_type partnerMcsUnit = 0;
 
+	// Add temp code to handle different MCS pairing
+	if (startMcsUnit == 0)
+	{
+		partnerMcsUnit = 4;
+	}
+	else if (startMcsUnit == 1)
+	{
+		partnerMcsUnit = 5;
+	}
+	else if (startMcsUnit == 4)
+	{
+		partnerMcsUnit = 0;
+	}
+	else if (startMcsUnit == 5)
+	{
+		partnerMcsUnit = 1;
+	}
     // If CHIP_UNIT is even, its partner will be the next MCS
-    if (!(startMcsUnit % 2))
+    else if (!(startMcsUnit % 2))
     {
         partnerMcsUnit = startMcsUnit + 1;
     }
-- 
1.8.4.msysgit.0

